modules/table/src/integrationTest/java/org/apache/ignite/distributed/ItTablePersistenceTest.java

```diff
@@ -44,193 +47,305 @@
 
-    private final List<TxManager> managers = new ArrayList<>();
+    /** Map of node indexes to partition listeners. */
+    private final Map<Integer, PartitionListener> partListeners = new ConcurrentHashMap<>();
+
+    /** Map of node indexes to table storages. */
+    private final Map<Integer, MvTableStorage> mvTableStorages = new ConcurrentHashMap<>();
+
+    /** Map of node indexes to partition storages. */
+    private final Map<Integer, MvPartitionStorage> mvPartitionStorages = new ConcurrentHashMap<>();
+
+    /** Map of node indexes to transaction managers. */
+    private final Map<Integer, TxManager> txManagers = new ConcurrentHashMap<>();
+
+    private ReplicaService replicaService;
 
     private final Function<String, ClusterNode> consistentIdToNode = addr
             -> new ClusterNode("node1", "node1", new NetworkAddress(addr, 3333));
 
-    private final ReplicaService replicaService = mock(ReplicaService.class);
+    private final HybridClock hybridClock = new HybridClockImpl();
+
+    private int stoppedNodeIndex;
+
+    private InternalTable table;
+
+    private final LinkedList<AutoCloseable> closeables = new LinkedList<>();
 
     @BeforeEach
-    void initMocks() {
-        doReturn(CompletableFuture.completedFuture(null)).when(replicaService).invoke(any(), any());
+    @Override
+    public void beforeTest(TestInfo testInfo) {
+        super.beforeTest(testInfo);
+
+        closeables.clear();
     }
 
     @AfterEach
     @Override
     public void afterTest() throws Exception {
         super.afterTest();
 
-        for (TxManager txManager : managers) {
-            txManager.stop();
-        }
+        closeAll(closeables);
     }
 
     /** {@inheritDoc} */
     @Override
     public void beforeFollowerStop(RaftGroupService service, RaftServer server) throws Exception {
-        // TODO: https://issues.apache.org/jira/browse/IGNITE-17817 Use Replica layer with new transaction protocol.
-        TxManagerImpl txManager = new TxManagerImpl(replicaService, new HeapLockManager(), new HybridClockImpl());
+        PartitionReplicaListener partitionReplicaListener = mockPartitionReplicaListener(service);
+
+        replicaService = mock(ReplicaService.class);
 
-        managers.add(txManager);
+        when(replicaService.invoke(any(), any()))
+                .thenAnswer(invocationOnMock -> partitionReplicaListener.invoke(invocationOnMock.getArgument(1)));
 
-        txManager.start();
+        for (int i = 0; i <= 2; i++) {
+            TxManager txManager = new TxManagerImpl(replicaService, new HeapLockManager(), hybridClock);
+            txManagers.put(i, txManager);
+        }
 
-        var table = new InternalTableImpl(
+        table = new InternalTableImpl(
                 "table",
                 UUID.randomUUID(),
                 Int2ObjectMaps.singleton(0, service),
                 1,
                 consistentIdToNode,
-                txManager,
+                txManagers.get(0),
```

Comment:
is it ok that we pass here txManager with index 0, but node with this manager could be stopped?